mobile/lib/pages/editing/edit.page.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:immich_mobile/pages/editing/crop.page.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/widgets/common/immich_image.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+
+@immutable
+class EditImagePage extends StatelessWidget {
+  final Asset? asset;
+  final Image? image;
+
+  const EditImagePage({
+    super.key,
+    this.image,
+    this.asset,
+  }) : assert(
+            (image != null && asset == null) ||
+                (image == null && asset != null),
+            'Must supply one of asset or image');
+
+
+  @override
+  Widget build(BuildContext context) {
+    final ImageProvider provider = (asset != null)
+        ? ImmichImage.imageProvider(asset: asset!)
+        : (image != null)
+            ? image!.image
+            : throw Exception('Invalid image source type');
+
+    final Image imageWidget = (asset != null)
+        ? Image(image: ImmichImage.imageProvider(asset: asset!))
+        : (image != null)
+            ? image!
+            : throw Exception('Invalid image source type');
+
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.black,
+        leading: IconButton(
+          icon: const Icon(Icons.close_rounded, color: Colors.white, size: 24),
+          onPressed: () =>
+              Navigator.of(context).popUntil((route) => route.isFirst),
+        ),
+        actions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.done_rounded, color: Colors.white, size: 24),
+            onPressed: () async {
+              if (image == null) {
+                ImmichToast.show(
+                  durationInSecond: 1,
+                  context: context,
+                  msg: 'No edits made!',
+                  gravity: ToastGravity.BOTTOM,
+                );
```

Comment:
I wonder if this is necessary. If no edits were made, do we need to inform the user?

Reply:
Yea because google photos dont let people to save non edited images, which also makes sense here.